Homework/LoggerAsync/Services/FileService.cs

```
@@ -0,0 +1,95 @@
+using LoggerAsync.Helpers;
+using LoggerAsync.Services.Abstractions;
+
+namespace LoggerAsync.Services
+{
+    public class FileService : IFileService
+    {
+        public FileService()
+        {
+            var logDirectoryPath = LoggerHelper.GetLogDirPath();
+            
+            InitializeDirectoryAsync(logDirectoryPath).Wait();
+        }
+
+        private async Task InitializeDirectoryAsync(string path)
+        {
+            if (!Directory.Exists(path))
+            {
+                await CreateDirectoryAsync(path);
+            }
+        }
+
+        private static async Task CreateDirectoryAsync(string path)
+        {
+            try
+            {
+                await Task.Run(() => Directory.CreateDirectory(path));
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Error creating directory: {ex.Message}");
```

Better to throw an exception